pkg/util/k8sutil/k8sutil.go

```diff
@@ -423,12 +423,51 @@ func setupClientServiceURL(endpoint string) string {
 	return fmt.Sprintf("http://%s:2379", endpoint)
 }
 
+func setupInitContainerCommand(cs api.ClusterSpec, m *etcdutil.Member, service v1.Service) (string, error) {
+	DNSTimeout := defaultDNSTimeout
+	if cs.Pod != nil {
+		DNSTimeout = cs.Pod.DNSTimeoutInSecond
+	}
+	
+	host := ""
+	if cs.ClusteringMode == "discovery" {
+		serviceUrl, err := getServiceHostname(service)
+		if err != nil {
+			return "", err
+		}
+		host = serviceUrl
+	} else {
+		host = m.Addr()
+	}
+	return fmt.Sprintf(`
+		TIMEOUT_READY=%d
+		while ( ! nslookup %s )
+		do
+			# If TIMEOUT_READY is 0 we should never time out and exit
+			TIMEOUT_READY=$(( TIMEOUT_READY-1 ))
```

Comment:
I would suggest the use of time functions for circuit breaking like
https://pkg.go.dev/time#After
with start time and after just in case

Reply:
so instead of doing a bash logic in a init container we check this in etcd operator itself before creating the etcd pods?

Reply:
I think that for this version we can just use the init container, instead of having the operator do a dns resolution. Worst case scenario the pod will be forever in init state waiting for the dns resolution